contracts/modules/staking/StakingPoolFactory.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+pragma solidity =0.8.16;
+
+import "../../interfaces/IStakingPoolFactory.sol";
+import "./MinimalBeaconProxy.sol";
+
+contract StakingPoolFactory is IStakingPoolFactory {
+
+  address public operator;
+  uint96 internal _stakingPoolCount;
+
+  // temporary beacon address storage to avoid constructor arguments in the proxy
+  address public beacon;
+
+  constructor(address _operator) {
+    operator = _operator;
+  }
+
+  function stakingPoolCount() external view returns (uint) {
+    return _stakingPoolCount;
+  }
+
+  function create(address _beacon) external returns (uint poolId, address stakingPoolAddress) {
+
+    require(
+      msg.sender == operator,
+      "StakingPoolFactory: Only operator can create staking pools"
+    );
+
+    beacon = _beacon;
```

Comment:
Does it make sense to empty `beacon` after the proxy is deployed for the gas refund?

Reply:
It's a similar situation to the reentrancy locks - OZ decided to use values `1` and `2` instead of zero - non-zero. The reason is because the next time the slot will be dirty and it will be cheaper to write.